packages/next/src/server/app-render/create-component-tree.tsx

```diff
@@ -704,6 +798,28 @@ async function createComponentTreeInternal({
                 </>
               ) : undefined
             }
+            forbidden={
+              Forbidden ? (
+                <>
+                  {layerAssets}
+                  <SegmentComponent params={params}>
+                    {forbiddenStyles}
+                    <Forbidden />
+                  </SegmentComponent>
+                </>
+              ) : undefined
+            }
+            unauthorized={
+              Unauthorized ? (
+                <>
+                  {layerAssets}
+                  <SegmentComponent params={params}>
+                    {unauthorizedStyles}
+                    <Unauthorized />
+                  </SegmentComponent>
+                </>
+              ) : undefined
+            }
```

Comment:
can we do something about this per my todo above? feels very submoptimal

Reply:
Realized this is not necessary to have forbidden and unauthorized boundaries here as unmatched parallel will only throw notFound. That one is a tricky case for parallel routes I feel we should address it separately